src/constructRoutes.js

```diff
@@ -13,7 +13,9 @@ import { resolvePath } from "./matchRoute.js";
 import { find } from "./utils/find";
 
 /**
- * @typedef {InputRoutesConfigObject | HTMLElement | import('parse5').DefaultTreeDocument} RoutesConfig
+ * @typedef {Element | null} ElementOrNull
```

Comment:
I don't know how you'll feel about this, but this is because document.querySelector can return `null`. And that matches how the documentation is currently shows how to use it [`const routes = constructRoutes(document.querySelector('#single-spa-layout'));
`](https://single-spa.js.org/docs/layout-overview/)

Reply:
I think I'd prefer requiring the consumer of the library to ensure that the correct value is being passed in. Perhaps through type casting.